packages/select/src/Select/Select.tsx

```diff
@@ -64,10 +73,10 @@ const Select = <T extends boolean>({
       aria-label={label}
       isSearchable={isSearchable}
       menuPlacement={menuPlacement}
-      closeMenuOnSelect={false}
       hideSelectedOptions={false}
       unstyled
       menuPortalTarget={portalTarget}
+      {...(matchFrom === 'start' ? { filterOption: customSearchFilter } : {})}
```

Comment:
LGTM, men foreslår en liten refaktorering

```suggestion
      filterOption={matchFrom === 'start' ? customSearchFilter : undefined}
```